### ISSUE TITLE
Make commands more UNIX-like

### ISSUE THREAD

**Issue by arredond** [jsanz/carto_cli#13]
Many commands here (especially those that take a single argument) could benefit from a more UNIX-like approach, where arguments can be declared explicitly or passed in a pipe as `stdin`. For example, one could locate and kill queries in a single line:

```bash
carto_sql queries | jq '.[0].pid' | carto_sql kill
```

I tend to read a Batch query's status just after running it, in case it's failed. Instead of copying and pasting the job_id, one could just do:

```bash
carto_batch create "SELECT * FROM rivers LIMIT 10;" | carto_batch read
```

The approach is simple enough: check if arguments are passed and, if not, read from `stdin`. A nice way of avoiding having to wait for user input is detailed [in this blogpost](https://travelingfrontiers.wordpress.com/2010/06/10/creating-python-unix-style-command-line-scripts-how-to-avoid-sys-stdin-hanging-while-it-waits-for-input-from-the-keyboard/), which depends on checking whether it's an interactive session or not.

I gave this a small test on the later example and it works pretty well:

```python
@click.command(help="Returns details about a job id (JSON)")
@click.argument('job_id', required=False)
@click.help_option('-h', '--help')
@click.pass_context
def read(ctx,job_id):
    '''
    Just returns the details of a job using as a JSON
    '''
    carto_obj = ctx.obj['carto']
    if not job_id:
        if not sys.stdin.isatty(): # Don't wait for input
            job_id = sys.stdin.read()
        else:
            ctx.fail(
                "A job ID is required. "
                "Either pass it explicitly or pipe into the command"
            )
    try:
        job_details = carto_obj.batch_check(job_id)
        click.echo(json.dumps(job_details))
    except Exception as e:
        ctx.fail(e)
```

@jsanz let me know if you think this can be an interesting feature and I'd be happy to provide a PR. I think this could be useful for single-argument commands, such as:
- `carto_env load`
- `carto_env list`
- `carto_sql run`
- `carto_sql kill`
- `carto_batch create`
- `carto_batch read`
- most `carto_dataset` commands

**Comment by jsanz** [comment]
that'd be awesome @arredond, please go ahead I'll be more than happy to test and review it!

**Comment by arredond** [comment]
Done! The final solution (using callbacks) was completely based on [this Stackoverflow answer](https://stackoverflow.com/a/45845513/8159847). I mostly just added an error message.

**Comment by jsanz** [comment]
Thanks @arredond, I've released `0.0.4` with your improvements :rocket:
